### PR TITLE
Model adjustments for Themis

### PR DESCRIPTION
**Pull request by terlan98**
Involves changes required to integrate the shared models into Themis.

### CODE REVIEW

File: Sources/SharedModels/Exercise/Participation/TemplateParticipation.swift [hunk 1]
```diff
@@ -0,0 +1,13 @@
+//
+//  TemplateParticipation.swift
+//  
+//
+//  Created by Tarlan Ismayilsoy on 03.05.23.
+//
+
+import Foundation
+
+// Should ideally extend `ProgrammingExerciseStudentParticipation`, but can't since it's a struct. Can be left as it is for now
```
Comment (sven0311) [inline Sources/SharedModels/Exercise/Participation/TemplateParticipation.swift:10]: Maybe take a look at the StudentParticipation Protocol and solve it similarly. I don't know exactly about the hierarchy here on the serverside, but I think it's solvable in a better way

File: Sources/SharedModels/User/Account.swift [hunk 2]
```diff
@@ -6,6 +6,7 @@ public struct Account: Codable {
     public let login: String
     public let name: String
     public let firstName: String
+    public let lastName: String?
```
Comment (sven0311) [inline Sources/SharedModels/User/Account.swift:9]: Is it optional? Otherwise maybe make it non-optional to stay consistent with the other fields here

**Review by sven0311 — CHANGES_REQUESTED** [review]
Just some small remarks. If you like we can also have a quick call about the templateParticipation

File: Sources/SharedModels/User/Account.swift (hunk 2 above)
Comment (terlan98) [reply Sources/SharedModels/User/Account.swift:9]: It can be `nil` as a part of the `user` field contained in `ProgrammingExerciseStudentParticipation`

**Review by sven0311 — APPROVED** [review]
'APPROVE' performed by Xcode.

**Pull request merged**